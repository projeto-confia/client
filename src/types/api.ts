import { ImageProps } from './image'
import { Nullable } from './nullable'
import { SocialNetworkType } from './socialNetwork'

export type ButtonLinkProps = {
  label: string
  url: string
}

export type LandingPageHeaderProps = {
  title: string
  description: string
  button: ButtonLinkProps
  image: ImageProps
}

export type ProcessStepProps = {
  label: string
  description: string
  icon: ImageProps
}

export type SectionHowWorksProps = {
  title: string
  processSteps: ProcessStepProps[]
}

export type SectionAboutProps = {
  title: string
  subtitle: string
  description: string
  image: ImageProps
}

export type SponsorProps = {
  name: string
  logo: ImageProps
}

export type SectionSponsorsProps = {
  title: string
  sponsors: SponsorProps[]
}

export type SocialNetworkLinkProps = {
  socialNetwork: SocialNetworkType
  url: string
}

export type MemberProps = {
  name: string
  role: string
  photo: ImageProps
  socialNetworkLinks: Nullable<SocialNetworkLinkProps[]>
}

export type SectionTeamProps = {
  title: string
  members: MemberProps[]
}

export type StatisticDataProps = {
  title: string
  subtitle: string
  count: number
}

export type SectionStatisticsProps = {
  statisticDatas: StatisticDataProps[]
}

export type LandingPageProps = {
  logo: ImageProps
  header: LandingPageHeaderProps
  sectionStatistics: SectionStatisticsProps
  sectionHowWorks: SectionHowWorksProps
  sectionAbout: SectionAboutProps
  sectionSponsors: SectionSponsorsProps
  sectionTeam: SectionTeamProps
}
