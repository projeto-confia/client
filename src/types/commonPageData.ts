export type CommonPageDataProps = {
  tabTitle: string
}
