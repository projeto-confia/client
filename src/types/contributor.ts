import { ImageProps } from './image'
import { Nullable } from './nullable'
import { SocialNetworkLinkProps } from './socialNetworkLink'

export type ContributorProps = {
  id: string
  name: string
  role: string
  description: string
  photo: ImageProps
  socialNetworkLinks: Nullable<SocialNetworkLinkProps[]>
}
