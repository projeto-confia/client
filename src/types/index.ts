export * from './homePage'
export * from './nullable'
export * from './image'
export * from './socialNetwork'
export * from './publicationsPage'
export * from './publication'
export * from './sectionStatistics'
export * from './sectionOverview'
export * from './projectPage'
export * from './videoCard'
export * from './eventsPage'
