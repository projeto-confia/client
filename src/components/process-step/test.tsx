import { render, screen } from '@testing-library/react'
import { ProcessStepProps } from '../../types/index'

import ProcessStep from '.'

const DEFAULT_PROPS: ProcessStepProps = {
  label: 'Step 01',
  description: '<p>Some very descriptive description</p>',
  icon: {
    url: '/image.png',
    alternativeText: 'A not found image',
  },
}

describe('<ProcessStep />', () => {
  it('should render', () => {
    const { container } = render(<ProcessStep {...DEFAULT_PROPS} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render heading label', () => {
    render(<ProcessStep {...DEFAULT_PROPS} />)

    expect(
      screen.getByRole('heading', { name: DEFAULT_PROPS.label })
    ).toBeInTheDocument()
  })
})
