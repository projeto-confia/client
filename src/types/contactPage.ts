import { CommonPageDataProps } from './commonPageData'
import { SocialNetworkLinkProps } from './socialNetworkLink'

export type ContactPageProps = {
  commonPageData: CommonPageDataProps
  title: string
  email: string
  socialNetworkLinks: SocialNetworkLinkProps[]
}
