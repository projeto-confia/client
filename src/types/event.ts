import { CommonPageDataProps } from './commonPageData'
import { MediaProps } from './media'

export type EventProps = {
  commonPageData: CommonPageDataProps
  slug: string
  title: string
  subtitle?: string
  media: MediaProps
  content: string
}
