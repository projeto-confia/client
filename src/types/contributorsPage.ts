import { CommonPageDataProps } from './commonPageData'
import { ContributorProps } from './contributor'

export type ContributorsPageProps = {
  commonPageData: CommonPageDataProps
  title: string
  members: ContributorProps[]
}
