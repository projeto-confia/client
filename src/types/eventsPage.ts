import { CommonPageDataProps } from './commonPageData'
import { EventProps } from './event'

export type EventsPageProps = {
  commonPageData: CommonPageDataProps
  title: string
  events: EventProps[]
}
