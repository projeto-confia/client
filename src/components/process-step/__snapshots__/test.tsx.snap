// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProcessStep /> should render 1`] = `
<article
  class="MuiBox-root MuiBox-root-6 makeStyles-wrapper-1"
>
  <header
    class="makeStyles-header-2"
  >
    <h2
      class="MuiTypography-root makeStyles-label-3 MuiTypography-body1"
    >
      Step 01
    </h2>
  </header>
  <div
    class="MuiBox-root MuiBox-root-7 makeStyles-content-4"
  >
    <img
      alt="A not found image"
      class="makeStyles-image-5"
      src="undefined/image.png"
    />
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      &lt;p&gt;Some very descriptive description&lt;/p&gt;
    </p>
  </div>
</article>
`;
